module/zfs/vdev_indirect.c

```diff
@@ -770,8 +771,9 @@ spa_condense_indirect_start_sync(vdev_t *vd, dmu_tx_t *tx)
 	ASSERT(spa_feature_is_active(spa, SPA_FEATURE_OBSOLETE_COUNTS));
 	ASSERT(vdev_indirect_mapping_num_entries(vd->vdev_indirect_mapping));
 
-	uint64_t obsolete_sm_obj = vdev_obsolete_sm_object(vd);
-	ASSERT(obsolete_sm_obj != 0);
+	uint64_t obsolete_sm_obj;
+	VERIFY0(vdev_obsolete_sm_object(vd, &obsolete_sm_obj));
+	ASSERT3U(obsolete_sm_obj, !=, 0);
```

Comment:
Seems like we should either change `0` to `DMU_NO_OBJECT` in this context, or change vdev_obsolete_sm_object() to simply return `0`.